s4/clarity/configuration/protocol.py

```diff
@@ -123,6 +123,14 @@ def permitted_control_types(self):
         control_types = self.xml_findall("./permitted-control-types/control-type")
         return [ControlType(self.lims, p.get("uri")) for p in control_types]
 
+    @lazy_property
+    def permitted_instrument_types(self):
+        """
+        :type: List[str]
+        """
+        instrument_type_nodes = self.xml_findall("./permitted-instrument-types/instrument-type")
+        return list({node.text for node in instrument_type_nodes})
```

Comment:
Why wouldn't you just do `[node.text for node in instrument_type_nodes]`? Are you expecting duplicates?

Reply:
Not anymore :)

I started a shallow analysis of `Instruments` -- configuring an instrument in clarity to review `StepConfiguration`. So, I initially wrote this function anticipating a need to omit duplicate strings. 

However, I did later find that the Clarity UX/UI won't permit duplicate names (as you note below). Sadly, this developer didn't recall the initial commit, and failed to loop back/correct upon new learnings.